COVERAGE.md

```diff
@@ -0,0 +1,182 @@
+# Coverage testing
+
+smalltalkCI has built-in support for per-method coverage testing in Pharo and Squeak.
+It natively supports uploading coverage to [Coveralls](https://coveralls.io) from Travis CI or AppVeyor.
+
+Other CI services/Coverage reporters are supported by smalltalkCI with the [LCOV output format](#uploading-with-different-ci-services%2Fcoverage-reporters).
+
+## Configuring coverage testing
+
+To enable coverage testing behavior, add a `#coverage` dictionary to the `#testing` slot of your `.smalltalk.ston`:
+```javascript
+SmalltalkCISpec {
+  ...
+  #testing : {
+    ...
+    #coverage : {
+      #packages : [ 'Packages-To-Cover.*' ],
+      #classes : [ #ClassToCover, #'ClassToCover class' ],
+      #categories : [ 'Categories-To-Cover*' ]
+      #format : #coveralls
+    }
+  }
+}
+```
+The `#coverage` dictionary can contain the following options:
+- `#packages` (recommended)
+  - Measure coverage of all instance side methods in the provided packages
+- `#classes`
+  - Measures all methods of all provided classes (instance side)
+- `#categories`
+  - Measure coverage for all classes' methods as well as their meta classes' methods
+- `#format` (defaults to `#coveralls`)
+  - The output format of the Coverage data 
+  - May be either `#coveralls` or `#lcov`
+
+When running smalltalkCI on TravisCI or AppVeyor with the `#coveralls` coverage format, the results will be uploaded to [Coveralls](https://coveralls.io) automatically.
+Make sure your repository is [added to Coveralls](https://coveralls.io/repos/new).
+
+## Uploading with different CI-services/Coverage reporters
+To support as many combinantions of CI services and coverage reporters, smalltalkCI supports the [LCOV](http://ltp.sourceforge.net/coverage/lcov.php) coverage output format.
+
+When `#format` is set to `#lcov`, coveralls will write a file named `lcov.info` into the `${SMALLTALK_CI_BUILD}` directory.
+Most coverage services already support uploading coverage in the LCOV format with uploader scripts.
+
+For the most common usecases, see these instructions:
+- [Inspecting coverage locally](#inspecting-coverage-locally)
+- [Coveralls](#coveralls)
+  - [Travis CI](#coveralls-%26-travis-ci)
+  - [Github actions](#coveralls-%26-github-actions)
+- [CodeCov](#codecov)
+  - [Travis CI](#codecov-%26-travisci)
+  - [Github actions](#codecov-%26-github-actions)
+
+### Inspecting coverage locally
+On Linux distributions, LCOV is available as a set of tools that can generate a coverage report as HTML/CSS files.
+When running smalltalkCI with LCOV coverage enabled, the log will contain a message like this:
+```shell
+Writing LCOV coverage info to: /path/to/lcov.info
+```
+If you have LCOV installed, you can then generate the report:
+```bash
+cd /path/to
+genhtml lcov.info
+xdg-open index.html
+```
+The result will look something like this:
+
+<img src="resources/lcov-example.png" alt="An example LCOV report" width=75% />
```

Comment:
Could we please move the `lcov-example.png` to https://github.com/hpi-swa/smalltalkCI/tree/assets? It does not need to be part of the master branch, which is pulled every time a project is tested

Reply:
`lcov-example.png` has been moved to assets issue #467